Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/GetByFilter/GetStreetcodeByFilterHandler.cs

```diff
@@ -0,0 +1,135 @@
+            {
+                if(fact.Title.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+                {
+                    fact.Streetcodes.ForEach(streetcode => results.Add(CreateFilterResult(streetcode, fact.Title, "Wow-факти", "wow-facts")));
+                    continue;
+                }
+
+                if (fact.FactContent.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+                {
+                    fact.Streetcodes.ForEach(streetcode => results.Add(CreateFilterResult(streetcode, fact.FactContent, "Wow-факти", "wow-facts")));
+                }
+            }
+
+            foreach (var timelineItem in await _repositoryWrapper.TimelineRepository.GetAllAsync(include: i => i.Include(x => x.Streetcodes)))
+            {
+                if (timelineItem.Title.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+                {
+                    timelineItem.Streetcodes.ForEach(streetcode => results.Add(CreateFilterResult(streetcode, timelineItem.Title, "Хронологія", "timeline")));
+                    continue;
+                }
+
+                if (!string.IsNullOrEmpty(timelineItem.Description) && timelineItem.Description.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+                {
+                    timelineItem.Streetcodes.ForEach(streetcode => results.Add(CreateFilterResult(streetcode, timelineItem.Description, "Хронологія", "timeline")));
+                }
+            }
+
+            foreach (var streetcodeArt in await _repositoryWrapper.ArtRepository.GetAllAsync(include: i => i.Include(x => x.StreetcodeArts)))
+            {
+                if (!string.IsNullOrEmpty(streetcodeArt.Description) && streetcodeArt.Description.Contains(searchQuery, StringComparison.OrdinalIgnoreCase))
+                {
+                    streetcodeArt.StreetcodeArts.ForEach(art =>
+                    {
+                        if (art.Streetcode == null)
+                        {
+                            return;
+                        }
+
+                        results.Add(CreateFilterResult(art.Streetcode, streetcodeArt.Description, "Арт-галерея", "art-gallery"));
+                    });
+                    continue;
+                }
+            }
+
+            return results;
+        }
+
+        private StreetcodeFilterResultDTO CreateFilterResult(StreetcodeContent streetcode, string content, string? sourceName = null, string? blockName = null)
+        {
+            return new StreetcodeFilterResultDTO
+            {
+                StreetcodeId = streetcode.Id,
+                StreetcodeUrl = $"http://localhost:3000/streetcode/{streetcode.TransliterationUrl}" + (string.IsNullOrEmpty(blockName) ? "" : $"#{blockName}"),
+                Content = content,
+                SourceName = $"Стріткод #{streetcode.Id:0000}" + (string.IsNullOrEmpty(sourceName) ? "" : $"/{sourceName}"),
+            };
```

Comment:
the number in the main card with # is not Id, it is the index property.
Also it is not good to set the URL here. you need just send the transliteration URL for streetcode, and the redirect set in the front-end, because now, you are closely tied with the URL for the website.

Reply:
also, i think the string better to form in the front, so as a response send only information about search such as streetcode id, index, URL, and all that you want else.

Reply:
done